examples/s3/bucketpolicy.yaml

```diff
@@ -21,12 +21,17 @@ spec:
         resource:
           - "arn:aws:s3:::test-bucket"
         condition:
+          StringEquals:
```

Comment:
When I look at the [docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html) you referenced, I see the following example:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": "dynamodb:GetItem",
            "Resource": "arn:aws:dynamodb:*:*:table/Thread",
            "Condition": {
                "ForAllValues:StringEquals": {
                    "dynamodb:Attributes": [
                        "ID",
                        "Message",
                        "Tags"
                    ]
                }
            }
        }
    ]
}
```

If I'm not mistaken, `Condition` here is a map of maps, i.e. the value of `ForAllValues:StringEquals` also seems to be a map object with single key `dynamodb:Attributes`. But in this PR, we have only one level map and then we use an array with `key` field. Was this intentional?

Kubernetes API patterns actually [discourage](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps) the use of maps whose value type is `struct`. On the other hand, we're trying to stick to how provider wants to represent their data structures. I feel closer to avoiding the maps with struct value and go with arrays but I can be convinced either way. Though I believe both levels should be either map or array instead of array inside of a map.

Reply:
Each object in the array represents a key/value pair under the "condition operator" (the map string key). I wasn't sure if there was a better way to represent this design. I will look into switching the design to use arrays on both sides.